mne/simulation/source.py

```diff
@@ -507,11 +530,13 @@ def get_stc(self, start_sample=0, stop_sample=None):
 
     def __iter__(self):
         """Iterate over 1 second STCs."""
-        # Arbitrary chunk size, can be modified later to something else
+        # Arbitrary chunk size, can be modified later to something else.
         # Loop over chunks of 1 second - or, maximum sample size.
         # Can be modified to a different value.
-        n_times = self.n_times
-        for start_sample in range(0, n_times, self._chk_duration):
-            stop_sample = min(start_sample + self._chk_duration, n_times)
+        last_sample = self.last_samp
+        for start_sample in range(self.first_samp, last_sample + 1,
```

Comment:
There is a failure in CircleCI here:

https://app.circleci.com/pipelines/github/mne-tools/mne-python/7473/workflows/c20954f6-74e6-4e15-b084-d0b8c26badc2/jobs/26531

Can you look into it @NeuroLaunch ?